python/lsst/meas/extensions/shapeHSM/_hsm_shape.py

```diff
@@ -0,0 +1,346 @@
+        # (here int32).
+        badpix = galsim.Image(badpix, bounds=bounds, copy=False)
+
+        # FIXME: dummyMask not used later on?! (it was in C++!)
+        # dummyMask = galsim.ImageI(bounds=bounds)
+        # dummyMask.image <- 1  # use something to make all pixel values 1
+
+        sctrl = afwMath.StatisticsControl()
+        sctrl.setAndMask(bitValue)
+
+        # NOTE: Origin defaults to PARENT in all cases accessible from Python.
+        variance = afwImage.Image(
+            exposure.variance[bbox],
+            dtype=exposure.variance.dtype,
+            deep=False,
+        )
+        stat = afwMath.makeStatistics(variance, subMask, afwMath.MEDIAN, sctrl)
+
+        skyvar = stat.getValue(afwMath.MEDIAN)
+
+        try:
+            shape = galsim.hsm.EstimateShear(
+                gal_image=image,
+                PSF_image=psf,
+                weight=None,
+                badpix=badpix,
+                sky_var=skyvar,
+                shear_est=self.config.shearType,
+                recompute_flux="FIT",
+                guess_sig_gal=2.5 * psfSigma,
+                guess_sig_PSF=psfSigma,
+                precision=1.0e-6,
+                guess_centroid=galsim.PositionD(center.getX(), center.getY()),
+                strict=True,
+                hsmparams=None,
+            )
+        except galsim.hsm.GalSimHSMError as error:
+            raise measBase.MeasurementError(str(error), self.GALSIM.number)
+
+        if shape.meas_type != self.measTypeSymbol:
+            # Give up on the plugin entirely without wasting time measuring
+            # each source and failing every single time.
+            raise measBase.FatalAlgorithmError("The plugin setup is incompatible with GalSim.")
+
+        if shape.meas_type == "e":
+            record.set(self.e1Key, shape.corrected_e1)
+            record.set(self.e2Key, shape.corrected_e2)
+            record.set(self.sigmaKey, 2.0 * shape.corrected_shape_err)
+        else:
+            record.set(self.e1Key, shape.corrected_g1)
+            record.set(self.e2Key, shape.corrected_g2)
+            record.set(self.sigmaKey, shape.corrected_shape_err)
+
+        record.set(self.resolutionKey, shape.resolution_factor)
+        self.flagHandler.setValue(record, self.FAILURE.number, shape.correction_status != 0)
+
+    def fail(self, record, error=None):
+        # Docstring inherited.
+        self.flagHandler.handleFailure(record)
+        if error:
+            centroid = self.centroidExtractor(record, self.flagHandler)
+            self.log.debug(
+                "Failed to measure shape for %d at (%f, %f): %s",
+                record.getId(),
+                centroid.getX(),
+                centroid.getY(),
+                error,
+            )
+
+
+class HsmShapeBjConfig(HsmShapeConfig):
+    """Configuration for HSM shape measurement for the BJ estimator."""
+
+    def setDefaults(self):
+        super().setDefaults()
+        self.shearType = "BJ"
+
```

Comment:
I'd add:

```python
def validate(self):
    super().validate()
    if self.shearType != "BJ":
        raise measBase.FatalAlgorithmError("Incompatible shearType")
```

(or something like that; maybe there's a better exception class to use)...

and similar down below.

Reply:
Yeah, and it's a good practice to call the `super().validate()` at the end of the method rather at the beginning.

Reply:
Done!